diff --git a/src/asm_file.py b/src/asm_file.py
index 2eab66e..42260a2 100644
--- a/src/asm_file.py
+++ b/src/asm_file.py
@@ -259,9 +259,16 @@ def parse_incbin(
     args: List[str], options: Options, warnings: List[str]
 ) -> Optional[bytes]:
     try:
-        filename = args[0]
-        offset = int(args[1], 0)
-        size = int(args[2], 0)
+        if len(args) == 3:
+            filename = args[0]
+            offset = int(args[1], 0)
+            size = int(args[2], 0)
+        elif len(args) == 1:
+            filename = args[0]
+            offset = 0
+            size = -1
+        else:
+            raise ValueError
     except ValueError:
         raise DecompFailure(f"Could not parse asm_data .incbin directive: {args}")
 
@@ -283,7 +290,7 @@ def parse_incbin(
         except MemoryError:
             data = b""
 
-        if len(data) != size:
+        if (len(data) != size) and (size > 0):
             add_warning(
                 warnings,
                 f"Unable to read {size} bytes from {full_path} at {offset:#x} (got {len(data)} bytes)",
@@ -293,7 +300,7 @@ def parse_incbin(
 
     add_warning(
         warnings,
-        f"Unable to find {filename} in any of {len(options.incbin_dirs)} search paths",
+        f"Unable to find {full_path} in any of {len(options.incbin_dirs)} search paths",
     )
     return None
 
diff --git a/src/main.py b/src/main.py
index 69c3ec5..058ebbc 100644
--- a/src/main.py
+++ b/src/main.py
@@ -76,6 +76,8 @@ def run(options: Options) -> int:
     asm_data = AsmData()
     try:
         for filename in options.filenames:
+            if filename == "data/rsp.s":
+                continue
             if filename == "-":
                 asm_file = parse_file(sys.stdin, arch, options)
             else:
